lerobot/common/policies/vla/modeling_vla.py

```diff
@@ -78,299 +66,210 @@ def reset(self):
+                        {"type": "text", "text": text},
+                        {"type": "image"},
+                    ],
+                },
+            ]
+            prompt = self.processor.apply_chat_template(
+                conversation, add_generation_prompt=add_generation_prompt
+            )
+        elif "paligemma" in self.vlm_backbone_name:
+            prompt = f"<image>{text}"
+        else:
+            prompt = text
+
+        return prompt
+
+    def get_vlm_features(self, processed_inputs) -> torch.Tensor:
+        vlm_output = self.vision_language_model(
+            **processed_inputs, return_dict=True, output_hidden_states=True
+        )
 
-    def forward(self, processed_inputs, batch_size):
+        if any(k in self.vlm_backbone_name for k in ["llava-onevision", "paligemma"]):
+            batch_size = processed_inputs["input_ids"].shape[0]
+            last_hidden_state = vlm_output.hidden_states[-1]
+            seq_len = vlm_output.image_hidden_states.shape[0] // batch_size
+            image_features = vlm_output.image_hidden_states.view(batch_size, seq_len, -1)
+
+            if self.vlm_backbone_feature_selection == "first_image":
+                num_img_feats = 598  # this is specific to llava-onevision
+                hidden_states = image_features[:, :num_img_feats, :]
+            elif self.vlm_backbone_feature_selection == "last_token":
+                hidden_states = last_hidden_state[:, -1:, :]
+            elif self.vlm_backbone_feature_selection == "all_generated":
+                hidden_states = last_hidden_state
+            elif self.vlm_backbone_feature_selection == "all":
+                hidden_states = torch.cat((image_features, last_hidden_state), dim=1)
+            else:
+                raise NotImplementedError(" not supportedd")
+        else:
+            raise NotImplementedError(f"{self.vlm_backbone_name} not implemented.")
+
+        return hidden_states
+
+    def get_action_logits(self, hidden_states: torch.Tensor) -> torch.Tensor:
+        batch_size, _, hidden_size = hidden_states.shape
+        hidden_states = hidden_states.transpose(0, 1)
+        if self.action_decoder_name == "act":
+            # Generate positional embeddings for the decoder
+            decoder_pos_embeddings = self.decoder_pos_embed.weight.unsqueeze(1).repeat(1, batch_size, 1)
+            # Decode the action with positional embeddings and encoder output
+            x = torch.zeros(
+                (self.chunk_size, batch_size, hidden_size),
+                dtype=hidden_states.dtype,
+                device=hidden_states.device,
+            )
+            action_logits = self.action_decoder(
+                x=x, encoder_out=hidden_states, decoder_pos_embed=decoder_pos_embeddings
+            )
+            # Final action logits through the action head
+            action_logits = self.action_head(action_logits)
+            action_logits = action_logits.transpose(0, 1)
```

Comment:
that's looks good to me:) Indeed would be better if we want to add other decoders.